lib/trello/card.rb

```diff
@@ -491,31 +481,49 @@ def fields_has_key?(fields, key)
-      attributes[:board_id]               = fields[SYMBOL_TO_STRING[:board_id]] || attributes[:board_id]
-      attributes[:member_ids]             = fields[SYMBOL_TO_STRING[:member_ids]] || fields[:member_ids] || attributes[:member_ids]
-      attributes[:list_id]                = fields[SYMBOL_TO_STRING[:list_id]] || fields[:list_id] || attributes[:list_id]
-      attributes[:pos]                    = fields[SYMBOL_TO_STRING[:pos]] || fields[:pos] || attributes[:pos]
-      attributes[:labels]                 = (fields[SYMBOL_TO_STRING[:labels]] || []).map { |lbl| Trello::Label.new(lbl) }.presence || attributes[:labels].presence || []
-      attributes[:card_labels]            = fields[SYMBOL_TO_STRING[:card_labels]] || fields[:card_labels] || attributes[:card_labels]
-      attributes[:last_activity_date]     = Time.iso8601(fields[SYMBOL_TO_STRING[:last_activity_date]]) rescue nil if fields.has_key?(SYMBOL_TO_STRING[:last_activity_date])
-      attributes[:cover_image_id]         = fields[SYMBOL_TO_STRING[:cover_image_id]] || attributes[:cover_image_id]
-      attributes[:badges]                 = fields[SYMBOL_TO_STRING[:badges]] || attributes[:badges]
-      attributes[:card_members]           = fields[SYMBOL_TO_STRING[:card_members]] || attributes[:card_members]
-      attributes[:source_card_id]         = fields[SYMBOL_TO_STRING[:source_card_id]] || fields[:source_card_id] || attributes[:source_card_id]
-      attributes[:source_card_properties] = fields[SYMBOL_TO_STRING[:source_card_properties]] || fields[:source_card_properties] || attributes[:source_card_properties]
+      %i[
+        id short_id url short_url last_activity_date
+        badges card_members
+      ].each do |attr_key|
+        attributes[attr_key] = parse_readonly_fields(fields, attr_key)
+      end
+
+      %i[
+        name desc due due_complete closed board_id
+        member_ids list_id pos labels card_labels
+        cover_image_id source_card_id source_card_properties
+      ].each do |attr_key|
+        attributes[attr_key] = parse_writable_fields(fields, attr_key)
+      end
+
+      attributes[:last_activity_date] = Time.iso8601(attributes[:last_activity_date]) rescue nil if attributes[:last_activity_date]
+      attributes[:labels] = (attributes[:labels] || []).map { |label| label.kind_of?(Trello::Label) ? label : Trello::Label.new(label) }
+      attributes[:due_complete] ||= false
+      attributes[:due] = attributes[:due].is_a?(String) ? (Time.iso8601(attributes[:due]) rescue nil) : attributes[:due]
       self
     end
 
+    def parse_writable_fields(fields, key)
+      gem_version_key = key.to_sym
+      api_version_key = SYMBOL_TO_STRING[gem_version_key]
+      if fields.key?(api_version_key)
+        fields[api_version_key]
+      elsif fields.key?(gem_version_key)
+        fields[gem_version_key]
+      else
+        attributes[gem_version_key]
+      end
+    end
+
+    def parse_readonly_fields(fields, key)
```

Comment:
What do you think about setting a default value for `fields` param?

Reply:
Maybe I can set it here?

```ruby
def initialize_fields(fields)
  fields ||= {}
```

Is that ok?

Reply:
I added a default value for fields in both `parse_writable_fields` and `parse_readonly_fields` methods